File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if head is None:
            return head
        
        dummy = ListNode()
        dummy.next = head
        cur = dummy
        while cur is not None and cur.next is not None:
            if cur.next.val == val:    
                cur.next = cur.next.next
            else:
                cur = cur.next
        
        return dummy.next    
            
        
        

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
/*
func removeElements(head *ListNode, val int) *ListNode {
    dummy := &ListNode{
        Next: head,
    }
    cur := dummy
    for cur != nil && cur.Next != nil {
        if cur.Next.Val == val {
            cur.Next = cur.Next.Next
        } else {
            cur = cur.Next
        }
    }
    return dummy.Next
}
*/

func removeElements(head *ListNode, val int) *ListNode {
    if head == nil {
        return nil
    }
    head.Next = removeElements(head.Next, val)
    if head.Val == val {
        return head.Next
    }
    return head
}